### PR TITLE
Manage User Directory for workbench

### LINKED ISSUE #21592: Add directories manager widget for workbench
The common widgets library contains the directories manager widget. This can be ported to the workbench.

As with other wrappers from C++:
  * add the class `ManageUserDirectories` to the sip wrappers
  * add a python module in `mantidqt.widgets` that imports it
  * pull it in to the workbench and add a menu item under the `File` menu
  * a low priority task would be to see if you can come find a decent icon from https://fontawesome.com/icons. These icons can be accessed using the [`qtawesome`](http://qtawesome.readthedocs.io/en/stable/_generate/qtawesome.icon.html) python package that is available for to use

### PR DESCRIPTION
**Pull request by ElliotAOram**
**NOTE: This contains a merge of an older state of branch #21566 please ensure that #21584 is merged before this**

This PR adds the manager user directory to the workbench. It is now available from `File > Manage User Directories`. This implementation does the same as MantidPlot where multiple clicks on the `Manage User Directories` shows additional dialogs.

**Question for tester**
* Clicking on`Manage User Directories` creates additional dialog boxes. Do we want this behaviour or would it be better to refocus on the currently opened dialog box (if it is already open).

**To test:**

* Open workbench
* Open the `Manage User Directory` Dialog with through `File > Manage User Directories`
* Check it functions as expected (same as MantidPlot)

Fixes #21592

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by ElliotAOram** [comment]
This branch will need to be merged into master once #21566 is merged

**Comment by ElliotAOram** [comment]
A few options for icon:
* (re-purpose for sitemap) https://fontawesome.com/icons/sitemap?style=solid
* https://fontawesome.com/icons/folder?style=solid
* https://fontawesome.com/icons/folder-open?style=solid

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 1]
```diff
@@ -183,27 +197,41 @@ def set_splash(self, msg=None):
...
-                                    shortcut_context=Qt.ApplicationShortcut)
+                                    shortcut_context=Qt.ApplicationShortcut,
+                                    icon_name="fa.save")
+        action_manage_directories = create_action(self, "Manage User Directories",
+                                                  on_triggered=self.open_manage_directories,
+                                                  icon_name="fa.folder")
 
         action_quit = create_action(self, "&Quit", on_triggered=self.close,
                                     shortcut="Ctrl+Q",
                                     shortcut_context=Qt.ApplicationShortcut)
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: How about adding [`power-off`](https://fontawesome.com/icons/power-off?style=solid) for Quit?

**Comment by ElliotAOram** [comment]
looks like tests are failing on qtawesome import

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 2]
```diff
@@ -383,6 +473,10 @@ def save_file(self):
         # todo: how should this interact with project saving and workspaces when they are implemented?
         self.editor.save_current_file()
 
+    def open_manage_directories(self):
+        manage_directory_widget = ManageUserDirectories(self)
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: I'm not sure we should keep the behaviour of having multiple dialogs open at once; I don't think it makes sense. Let's use `._exec_()` and block until we are finished.

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Comment by martyngigg** [comment]
@ElliotAOram Thanks, I'd forgotten about getting `qtawesome` installed everywhere.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**